aws-lc-rs/src/agreement.rs

```diff
@@ -763,6 +838,52 @@ fn x25519_diffie_hellman<'a>(
     Ok(&buffer[0..AlgorithmID::X25519.pub_key_len()])
 }
 
+pub(crate) fn try_parse_x25519_public_key_bytes(
+    key_bytes: &[u8],
+) -> Result<LcPtr<EVP_PKEY>, Unspecified> {
+    if let Ok(key) = try_parse_x25519_subject_public_key_info_bytes(key_bytes) {
+        return Ok(key);
+    }
+    if let Ok(key) = try_parse_public_key_raw_bytes(key_bytes) {
+        return Ok(key);
+    }
+    Err(Unspecified)
+}
```

Comment:
This can be just a `or` invocation: 
````    
pub(crate) fn try_parse_x25519_public_key_bytes(
    key_bytes: &[u8],
) -> Result<LcPtr<EVP_PKEY>, Unspecified> {
    try_parse_x25519_subject_public_key_info_bytes(key_bytes)
        .or(try_parse_public_key_raw_bytes(key_bytes))
}
```